File: src/test/java/com/merricklabs/portlandbus/mocks/MockMyStopStorage.java
package com.merricklabs.portlandbus.mocks;

import com.google.inject.Singleton;
import com.merricklabs.portlandbus.storage.MyStopStorage;
import java.util.OptionalInt;

@Singleton
public class MockMyStopStorage implements MyStopStorage, TestMock {

    OptionalInt myStop = OptionalInt.empty();

    @Override
    public void saveStop(String userId, int stopId) {
        myStop = OptionalInt.of(stopId);
    }

    @Override
    public OptionalInt queryStopId(String userId) {
        return myStop;
    }

    @Override
    public void reset() {
        myStop = OptionalInt.empty();
    }
}

File: src/test/kotlin/com/merricklabs/portlandbus/mocks/MockMyStopStorage.kt
package com.merricklabs.portlandbus.mocks

import com.merricklabs.portlandbus.storage.MyStopStorage
import java.util.OptionalInt

class MockMyStopStorage : MyStopStorage, TestMock {

    private var myStop = OptionalInt.empty()

    override fun saveStop(userId: String, stopId: Int) {
        myStop = OptionalInt.of(stopId)
    }

    override fun queryStopId(userId: String): OptionalInt {
        return myStop
    }

    override fun reset() {
        myStop = OptionalInt.empty()
    }
}